XavierGuertin/WildfireVisualizationProject
# #74 - fix(Nx): Bug fix by transforming all nx config to local runner

### Summary
This PR removes the use of NX Cloud for local command execution and updates the E2E test configuration to ensure consistent test runs. Adjustments were made to improve local development workflows and address issues with the Cypress E2E test setup.

### Changes Made
- Switched from NX Cloud to local execution for build and test commands in `project.json`.
- Updated the E2E test configuration:
  - Configured Cypress to detect test files properly by setting the correct `cypressDir` path and `specPattern`.
  - Added `wait-on` to ensure the frontend server is ready before Cypress starts running tests.
  - Simplified the Cypress configuration to prevent redundant `ciBaseUrl` and `ciWebServerCommand` settings.
- Created a basic E2E test to verify the index page displays the text "Views" in a `span` element.

### Testing Instructions
1. Run `nx e2e frontend-e2e` to verify that the E2E tests execute successfully and fail if there is an issue.
2. Ensure that local commands (`nx build frontend`, `nx serve frontend`, etc.) execute without NX Cloud dependencies.
3. In CI/CD, confirm that Cypress runs headlessly and detects any failed tests, correctly passing or failing the pipeline as expected.

### Screenshots (if applicable)
_No screenshots required; changes are related to configuration and testing._

### Additional Notes
- Removed NX Cloud configurations for local tasks to reduce reliance on remote caching for development workflows.
- CI/CD should now detect Cypress test failures via non-zero exit codes, marking builds as failed if tests do not pass.
